### PR TITLE
Fix/state var init

### PR DESCRIPTION
**Pull request by correcthorsebatterystaple**
Fix state variable initialization throwing errors because of a faulty key access on the AST

### CODE REVIEW

**Review by lucasccordeiro — CHANGES_REQUESTED** [review]
@correcthorsebatterystaple: could you please add a regression test that produces `VERIFICATION FAILED`?

**Comment by lucasccordeiro** [comment]
BTW, the recently added regression test fails in our CI:

```
The following tests FAILED:
	727 - regression/esbmc-solidity/github_state_init (Failed)
Errors while running CTest
```

File: src/solidity-frontend/solidity_convert.cpp [hunk 1]
```diff
@@ -213,22 +215,22 @@ bool solidity_convertert::get_var_decl(
...
+  bool is_state_var = ast_node["stateVariable"] == true;
+
   // 2. populate id and name
   std::string name, id;
-  if(ast_node["stateVariable"] == true)
+  if(is_state_var)
     get_state_var_decl_name(ast_node, name, id);
+  else if(current_functionDecl)
+  {
+    // location variable in function body
```
Comment (kunjsong01) [inline src/solidity-frontend/solidity_convert.cpp]: local variable in function body?

File: regression/esbmc-solidity/github_state_init/test.desc [hunk 2]
```diff
@@ -0,0 +1,4 @@
+CORE
+contract.solast
```
Comment (kunjsong01) [inline regression/esbmc-solidity/github_state_init/test.desc]: The JSON AST file is missing in the commit.

**Review by kunjsong01 — CHANGES_REQUESTED** [review]

File: regression/esbmc-solidity/github_state_init/test.desc [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+--function state_var_init --contract contract.sol
+^VERIFICATION SUCCESSFUL$
```
Comment (kunjsong01) [inline regression/esbmc-solidity/github_state_init/test.desc]: Would it be possible to add another test case that reports "VERIFICATION FAILED"?

**Review by kunjsong01 — APPROVED** [review]

**Review by lucasccordeiro — APPROVED** [review]

**Pull request merged**